### PR TITLE
FIPS Validate Command

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR documents the `fips-validate` command that is available in the Palette CLI.

## Changed Pages

💻 [Preview URL](https://65eb2ca36e8185ab29799c2c--docs-spectrocloud.netlify.app/palette-cli/commands/fips-validate)

## Jira Tickets

🎫 [PLT-1080](https://spectrocloud.atlassian.net/browse/PLT-1080)

## Backports

Can this PR be backported?

- [ ] Yes. _Remember to add the relevant backport labels to your PR._
- [x] No. _Please leave a short comment below about why this PR cannot be backported._
Release PR.

[PLT-1080]: https://spectrocloud.atlassian.net/browse/PLT-1080?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docs-content/palette-cli/commands/fips-validate.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+- Sufficient permissions to create a new namespace and deploy resources in the namespace. We recommend using an elevated
+  ClusterRole such as [_cluster-admin_](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles)
+  to ensure that the command can create the necessary resources.
+
+- The Kubernetes cluster must be accessible from the machine where you issue the `fips-validate` command. Ensure that
+  the kubeconfig file is correctly configured to access the cluster.
+
+- The Kubernetes cluster must have internet access to download the images when using the `images` subcommand. Private
+  image registries are not supported.
+
```
Comment (addetz) [inline docs/docs-content/palette-cli/commands/fips-validate.md:55]:
```suggestion
- The Kubernetes cluster must have internet access to download the images when using the `images` subcommand.

```

Comment (addetz) [reply docs/docs-content/palette-cli/commands/fips-validate.md:55]: Removing as this belongs under limitations.

File: docs/docs-content/palette-cli/commands/fips-validate.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+The `images` subcommand accepts the following flags:
+
+| **Short Flag** | **Long Flag** | **Description**                                                                                        | **Type** |
+| -------------- | ------------- | ------------------------------------------------------------------------------------------------------ | -------- |
+| `-n`           | `--namespace` | The namespace in which to deploy the resources for the FIPS scan. The default namespace is `vertex-ns` | string   |
+| `-f`           | `--out`       | The output file path to save the scan results. The default output is the terminal.                     | string   |
+| `-h`           | `--help`      | Display the help message for the `images` subcommand.                                                  | boolean  |
+
+### Limitations
```
Comment (addetz) [inline docs/docs-content/palette-cli/commands/fips-validate.md:75]: I think all the limitations could be put under a single `## Limitations` section, which would be cleaner.

File: docs/docs-content/palette-cli/commands/fips-validate.md [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+:::info
+
+If you have the environment variable `KUBECONFIG` set, the `fips-validate` will automatically use the kubeconfig file
+and skip prompting you for the kubeconfig file path.
+
+:::
+
+### Supported Languages
+
+The `fips-validate` currently only supports Go binaries. Images with binaries compiled in other languages are marked as
```
Comment (addetz) [inline docs/docs-content/palette-cli/commands/fips-validate.md:24]: Should this just be a limitation?

File: docs/docs-content/palette-cli/commands/fips-validate.md [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+| `BINARY`   | The name of the executable binary.                                                                                                        |
+| `ARCH`     | The architecture of the binary.                                                                                                           |
+| `PATH`     | The path of the binary in the image.                                                                                                      |
+| `VERSION`  | The version of the programming language used to create the binary. Go is the only supported language from a version detection capability. |
+| `EXE`      | How the binary was compiled, either statically or dynamically.                                                                            |
+| `CRYPTO`   | The cryptographic library used to compile the binary.                                                                                     |
+
+The following is an example of the FIPS compliance report. The following image is FIPS compliant as the binary is
```
Comment (addetz) [inline docs/docs-content/palette-cli/commands/fips-validate.md:116]:
```suggestion
The following is an example of the FIPS compliance report. The validated image is FIPS compliant as the binary is
```

File: docs/docs-content/palette-cli/commands/fips-validate.md [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+compiled with the BoringSSL cryptographic library.
```
Comment (addetz) [inline docs/docs-content/palette-cli/commands/fips-validate.md:117]: Is this the only check it does? Asking as I don't see a similar checks table as under `services`

File: docs/docs-content/palette-cli/commands/fips-validate.md [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+│ palette-webhook-service.palette-system                         │ passed │
+│ 10-0-3-97.kubernetes.default:2380                              │ failed │
+│ 10-0-3-97.kubernetes.default:10250                             │ passed │
+└────────────────────────────────────────────────────────────────┴────────┘
+```
+
+## Clean
+
+The `clean` subcommand removes the FIPS validation resources from your clusters. The command is intended to be used in
+scenarios where the `images` or `services` subcommands fail to clean up the resources from the cluster. The `clean`
```
Comment (addetz) [inline docs/docs-content/palette-cli/commands/fips-validate.md:327]: Why would they fail? Perhaps they are still in use or the cluster is no longer healthy?

**Review by addetz — APPROVED** [review]
Such a cool tool! ⭐  Left a couple of questions as comments, but nothing major.

File: docs/docs-content/palette-cli/commands/fips-validate.md (hunk 5 above)
Comment (karl-cardenas-coding) [reply docs/docs-content/palette-cli/commands/fips-validate.md:116]: Yup, that's it

**Pull request merged**